### ISSUE TITLE
feat: check posted output root for fraud

### ISSUE THREAD

**Issue by ncitron** [a16z/magi#87]
Check the L1 to compare any output roots posted by the proposer to what we expect. If a mismatch is detected, very loudly warn the user via continuously posting warning messages to the CLI.

Might also be cool to have the Grafana dashboard display something if fraud is detected too.

related to #86

**Comment by clabby** [comment]
Wondering if it'd be a good idea to interop with the [op-challenger](https://github.com/clabby/op-challenger) here rather than add fault detection to magi? This service is meant to both detect and respond to fault in a post-permissionless output proposals world.

**Comment by clabby** [comment]
For some extra context, see the [challenger specification](https://github.com/ethereum-optimism/optimism/blob/develop/specs/challenger.md)

<img width="2061" alt="Permissionless Output Proposals" src="https://user-images.githubusercontent.com/8406232/234069292-fe9c3912-4d64-4168-a975-52b2f636ce6f.png">

**Comment by refcell** [comment]
I agree with @clabby here - it should be an isolated service. Exposing a trusted endpoint for faulty output detection would be high leverage here though.

**Comment by ncitron** [comment]
This makes a lot of sense @clabby @refcell. Agree that we should lean into the optimism-rs modularity and use op-challenger instead.

Can op-challenger be setup to not try and do the dispute game (since it doesn't exist yet if I understand correctly), and just alert if fraud is detected? I think that can still be useful as it would allow people to raise the alarm for the multisig to act in the case of the proposer being hacked for example. If so, I would love to integrate op-challenger into our docker configs once #102 is merged and update our grafana dashboard to freak out if fraud is detected.

**Comment by refcell** [comment]
That  sounds like a great idea. If it's not already configurable, i think it would make sense to add a "watch or listen mode" to the challenger. Thoughts @clabby ?
